### DISCUSSION TITLE
How does this compare to gsplat.js

### DISCUSSION THREAD

**Discussion by bmikaili** [mkkellogg/GaussianSplats3D#107] [General]
Has anyone tried both this and [Huggingface's gsplat.js](https://github.com/huggingface/gsplat.js)? How do they compare in terms of featureset and performance?

**Comment by mkkellogg** [comment]
I don't have great insights into the inner workings of `gsplat.js` myself so I'll leave it up to the community to respond. I'm also going to convert this to a discussion, which seems more appropriate

**Reply by bmikaili** [reply]
Wasn‘t aware of that feature, thanks!

**Comment by simonbethke** [comment]
I have tried both. gsplat appears to come with less overhead as it doesn't include three js. However, the performance seems to be comparable. Maybe gsplat is a bit better in modifying gsplats while this library takes all the power from three.js.

With three.js it appears to me that this might be a sweet-spot: Support for WebXR, other 3d-objects, and a good performance but not as fat as Babylon

**Comment by JeS24** [comment]
In addition to @simonbethke's points, as far as I am aware, `gsplat.js` does not support `.ply / .splat` streaming yet, unlike this library, which adds so much to the page's apparent responsiveness (mainly in terms of initial load time). The overall performance is on par though. Also, the unified API (`addSplatScene()`) for different file formats makes the code slightly simpler, as opposed to `gsplat`'s `PLYLoader()` and `Loader()` methods. And, from what I see, this library will probably add support for view-dependence (SH coefficients) before `gsplat.js`.

As for negatives, so far, the only downside has been the lack of a CDN version for `GaussianSplats3D`. `gsplat.js` is available via [jsdelivr](https://cdn.jsdelivr.net/npm/gsplat@latest).
